### PR TITLE
Celestia integration

### PR DESCRIPTION
**Pull request by zapnap**
Adds support for the [Celestia](https://celestia.org/) blockchain

### CODE REVIEW

**Comment by hodgerpodger** [comment]
Thanks for this great contribution!  

I'll merge after a little while.  Totally unrelated but working on redoing part of solana CSV (technical debt that has come to roost affecting deployment of everything on stake.tax).

**Comment by zapnap** [comment]
Sounds good! Lmk if you have any feedback on it when you go to merge it. And thanks for your ongoing work on stake.tax! It is a fantastic resource.

File: src/staketaxcsv/common/ibc/handle.py
Comment (hodgerpodger) [inline src/staketaxcsv/common/ibc/handle.py:1]: 70 internal tests break for staking transactions for juno/kuji/..., I think because of this file change.  Perhaps handling celestia staking transactions uniquely may be a solution instead of modifying common. 

I'm guessing there's a solution involving modifying common but I don't fully understand celestia data difference to know what that is.

**Review by hodgerpodger — CHANGES_REQUESTED** [review]
https://github.com/hodgerpodger/staketaxcsv/pull/318/files#r1412634246

File: src/staketaxcsv/common/ibc/handle.py
Comment (hodgerpodger) [reply src/staketaxcsv/common/ibc/handle.py:1]: Example of test breaking, where there's an extra "sent OSMO transaction" for a staking transaction in which there should be only one "receive OSMO".

Correct result
-------------------  -------  ---------------  -----------------  -----------  -------------  ---  ------------  ------------------------------------------------------------------
timestamp            tx_type  received_amount  received_currency  sent_amount  sent_currency  fee  fee_currency  txid
2022-07-25 20:05:35  STAKING  0.109096         OSMO                                                              XXXXXXXXXXXXXXXXXXXXXXXXXXX-0
-------------------  -------  ---------------  -----------------  -----------  -------------  ---  ------------  ------------------------------------------------------------------

Code result (incorrect)
-------------------  -------  ---------------  -----------------  -----------  -------------  ---  ------------  ------------------------------------------------------------------
timestamp            tx_type  received_amount  received_currency  sent_amount  sent_currency  fee  fee_currency  txid
2022-07-25 20:05:35  STAKING  0.109096         OSMO                                                              XXXXXXXXXXXXXXXXXXXXXXXXXXX-0
2022-07-25 20:05:35  STAKING                                      0.109096     OSMO                              XXXXXXXXXXXXXXXXXXXXXXXXXXX-0
-------------------  -------  ---------------  -----------------  -----------  -------------  ---  ------------  ------------------------------------------------------------------

Comment (zapnap) [reply src/staketaxcsv/common/ibc/handle.py:1]: It's a bit difficult to make any changes here to satisfy tests without a unit test suite to run against... It doesn't seem to be present in the repo. Can you share it?

Comment (zapnap) [reply src/staketaxcsv/common/ibc/handle.py:1]: may be worth finding a set of "unrelated" public addresses to specify for those tests? it's all a public ledger after all so 🤷‍♂️ I was obviously testing against my own (ad-hoc, manually) but having a standardized shared set with published unit tests would prevent regressions and also open it up to more contributors (and presumably lessen some of the load on you as maintainer)

Comment (hodgerpodger) [reply src/staketaxcsv/common/ibc/handle.py:1]: Ah okay--perhaps documentation saying addresses / transactions are randomly chosen from explorers like mintscan so it's clear, so that I can start building a public test suite.  Thanks.

For "extra sent transaction" issue, here's a random sample:

`python staketaxcsv/report_osmo.py osmo1jdx4rukctmdw7ad6xw5zjrlqchyj4kt632t2rn --txid DBB87D066C7052E36AB8A5BD4035F7270A4ABE615158DEFF4CEF3E30E3F84FB8`

(sample taken from  https://www.mintscan.io/osmosis/validators/osmovaloper1u5v0m74mql5nzfx2yh43s2tke4mvzghr6m2n5t , which are osmosis transactions involving using ecostake validator)

**Comment by hodgerpodger** [comment]
fyi i haven't neglected this.  I'm working on something that I hope supplants this (using mintscan api instead of lcd nodes for data; celestia is on mintscan api list; main motivation is that lcd nodes' history tends to be good for birth of blockchain's 1st year but over time, become unreliable in terms of archived data availability if the blockchain have no traction)

**Comment by hodgerpodger** [comment]
Thanks for the PR @zapnap !

I added celestia CSVs with https://github.com/hodgerpodger/staketaxcsv/commit/4be4b543de3cd559fc8db0f10ac906f8c2763b1e , using your PR as the guide.  

I didn't incorporate the changes for handle_staking.py, since I significant doubts there.

**Comment by zapnap** [comment]
great! looks like you sorted out some of the test suite issues as well? I have some other cosmos ecosystem updates in my local repo. If there's a published test suite now I'll rebase against it and try to get some other PRs in queue

**Comment by hodgerpodger** [comment]
Yup, I started a test suite (though just starting).  I'll likely migrate existing tests to the public test suite after tax season, since I'd like to get out a few things.  For now I mostly add public tests for new code.

**Pull request closed without merging**
